app/helpers/foreman_host_reports/reports_dashboard_helper.rb

```diff
@@ -54,26 +71,65 @@ def latest_reports?
       latest_reports.limit(1).present?
     end
 
-    def host_reports_translated_header(shortname, longname)
-      "<th style='width:85px; class='ca'><span class='small' title='' data-original-title='#{longname}'>#{shortname}</span></th>"
+    def failure_hosts_query(format)
+      "report_failures > 0 and report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"
     end
 
-    private
+    def failure_hosts(format)
+      Host.search_for(failure_hosts_query(format)).reorder("").count
+    end
 
-    def disabled_hosts(_format = nil)
-      Host.authorized(:view_hosts, Host).where(:enabled => false).count
+    def change_hosts_query(format)
+      "report_failures = 0 and report_changes > 0 and report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"
     end
 
     def change_hosts(format)
-      HostReport.authorized(:view_host_reports).my_reports.where("format = ? and change > 0", HostReport.formats[format]).reorder('').group(:host_id).maximum(:id).count
+      Host.search_for(change_hosts_query(format)).reorder("").count
+    end
+
+    def nochange_hosts_query(format)
+      "report_failures = 0 and report_changes = 0 and report_nochanges > 0 and report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"
     end
 
     def nochange_hosts(format)
-      HostReport.authorized(:view_host_reports).my_reports.where("format = ? and nochange > 0", HostReport.formats[format]).reorder('').group(:host_id).maximum(:id).count
+      Host.search_for(nochange_hosts_query(format)).reorder("").count
     end
 
-    def failure_hosts(format)
-      HostReport.authorized(:view_host_reports).my_reports.where("format = ? and failure > 0", HostReport.formats[format]).reorder('').group(:host_id).maximum(:id).count
+    def noevents_hosts_query(format)
+      "report_failures = 0 and report_changes = 0 and report_nochanges = 0 and report_format = #{format} and last_report > #{interval(format)} and status.enabled = true"
+    end
+
+    def noevents_hosts(format)
+      Host.search_for(noevents_hosts_query(format)).reorder("").count
+    end
+
+    def noreports_hosts_query(_format)
+      # always across all formats because there are no reports (thus format is unknown)
+      "null? last_report"
+    end
+
+    def noreports_hosts(format)
+      Host.search_for(noreports_hosts_query(format)).reorder("").count
+    end
+
+    def outofsync_hosts_query(format)
+      "report_format = #{format} and last_report <= #{interval(format)} and status.enabled = true"
+    end
+
+    def outofsync_hosts(format)
+      Host.search_for(outofsync_hosts_query(format)).reorder("").count
+    end
+
+    def disabled_hosts_query(format)
+      "report_format = #{format} and status.enabled = false"
+    end
+
+    def disabled_hosts(format)
+      Host.search_for(disabled_hosts_query(format)).reorder("").count
+    end
+
+    def total_hosts(format)
```

Comment:
I am not sure about that, which hosts do you want to count here? 
because to me it is  a bit confusing: 
![image](https://user-images.githubusercontent.com/57755873/160633393-6bd4fb5e-2360-4f82-9e73-230d5592a5b4.png)

Reply:
Total hosts = sum of all numbers without "across formats" typically (in a production deployment that is correctly configured). But this rule only applies when you don't have multiple reports for a host in the outofsync period (typically 30 minutes). See, reports come in intervals of typically 30 minutes (configurable), that is also the outofsync period (timeframe when Foreman considers reports as "expired" or "no longer recent"). If you upload/import/create many reports with various results the numbers will be skewed. In your case what happened is that you have a single host but you created/imported/migrated 4 reports with the reported_at/last_report within the outofsync timeframe. They are all reported.

Now, this can be solved by performing join query and iterating through results eliminating those invalid entries, thing is, we want to make search link to host list page and there is only scoped_search available.

We can theoretically solve that but that would be much more work which is I think not worth the effort. In the legacy reports, we do exactly the same, see `app/views/dashboard/_status_links.html.erb`. It is not ideal but this is a reasonable solution I think.